### PR TITLE
feat(rpc):  Add `tx_hash` to `debug_traceBlockByX` 

### LINKED ISSUE #5742: no txHash field in result for debug_traceBlockByNumber
### Describe the bug

Geth recently added the `txHash` in the results of `debug_traceBlockByNumber` https://github.com/ethereum/go-ethereum/pull/27183

### Steps to reproduce

1. Send `{"method":"debug_traceBlockByNumber","params":["latest", {"tracer": "callTracer"}],"id":1,"jsonrpc":"2.0"}` to the node
2. `txHash` not present in the output

### Node logs

_No response_

### Platform(s)

Linux (x86)

### What version/commit are you on?

reth 0.1.0-alpha.13 (c4c07b86)

### What database version are you on?

Current database version: 1

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

_No response_

### If you've built Reth from source, provide the full command you used

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by itsdevbear**
Closes #5742

### CODE REVIEW

**Review by Evalir — APPROVED** [review]

File: crates/rpc/rpc-types/src/eth/trace/common.rs [hunk 1]
```diff
@@ -8,7 +9,7 @@ use serde::{Deserialize, Serialize};
 #[allow(missing_docs)]
 pub enum TraceResult<Ok, Err> {
     /// Untagged success variant
-    Success { result: Ok },
+    Success { result: Ok, tx_hash: TxHash },
     /// Untagged error variant
     Error { error: Err },
```
Comment (tcoratger) [inline crates/rpc/rpc-types/src/eth/trace/common.rs:14]: If we look at the Go variant of that:

https://github.com/ethereum/go-ethereum/pull/27183/files#diff-bd288cd51c1874c7e65ce0d35047c0a98366479c8924606c9efc577f6a98f75c

it seems that the transaction hash information should also be added to the `Error` variant.

Comment (Evalir) [reply crates/rpc/rpc-types/src/eth/trace/common.rs:14]: correct, missed this comment—we need to add it to the error variant too

**Review by Evalir — DISMISSED** [review]

**Comment by itsdevbear** [comment]
@Evalir addressed

**Review by mattsse — APPROVED** [review]
lgtm,
I made the tx_hash type optional because this might not be supported by all clients

**Comment by itsdevbear** [comment]
Fire that makes sense

**Pull request merged**